### DISCUSSION TITLE
Testing components within another application (without going through tag/export)

### DISCUSSION THREAD

**Discussion by AlexEBall** [teambit/bit#4633] [Help] [answered]
Wonder if anyone from Bit team or the community has come across this situation. 

We have several collections on Bit (App A) and we want other apps (App B...  App n) to consume these components, but sometimes we want to test the App A components to make sure they're behaving correctly in the context of App B. Is there a way to connect App A components (or repo) with App B so that it allows for a quicker feedback loop while testing (i.e, going through several iterations, possibly modifying styles, logic, etc.) instead of modifying a component on App A tagging/exporting it to Bit, bumping the version in App B, `npm i` so that it can be consumed and updates be seen in App B?

@GiladShoham this was brought over from #technical-questions in Slack

**Comment by itaymendel** [comment, marked as answer]
hi @AlexEBall ,
there are two things you can do evaluating components before exporting:

### Using `compositions` as "liability testing"

When there are certain cases where you want to test a component against a very specific use case, you can create a `composition` that mimics the page/flow/state/etc of the component for a specific use case (note - you can have several `*.compositions.tsx` files for a single component, just make sure to use different named exports). This way any developer maintaining that component knows they might "break" an important use-case.

You can also test that composition by adding a `spec` file.

Internally we use this capability extensively.

### `bit link --target` (experimental)

You can use [`bit link --target`](https://harmony-docs.bit.dev/reference/cli-reference#link) to link the compiled output of a local component in a target project. Note, the feature is still experimental, so we'd appreciate feedbacks / bug reports.
